File: Easy/1422. Maximum Score After Splitting a String.java
class Solution {
    public int maxScore(String s) {
        final char[] sToChars = s.toCharArray();
        final int n = sToChars.length;
        int curScore = (sToChars[0] == '0') ? 1 : 0;
        for (int i = 1; i < n; i++) {
            curScore += (sToChars[i] == '0') ? 0 : 1;
        }
        int maxScore = curScore;
        for (int i = 1; i < n - 1; i++) {
            curScore += (sToChars[i] == '0') ? 1 : -1;
            maxScore = Math.max(curScore, maxScore);
        }
        return maxScore;
    }
}

File: Easy/1422. Maximum Score After Splitting a String.py
class Solution:
    def maxScore(self, s: str) -> int:
        curr_score, max_score = s.count('1'), 0
        for i in range(len(s) - 1):
            curr_score += 1 if s[i] == '0' else -1
            max_score = max(curr_score, max_score)
        return max_score